packages/models/src/errors.ts

```diff
@@ -102,46 +105,60 @@ export function makeApiProblemBuilder<T extends string>(errors: {
       })),
     });
 
-    return (
-      match<unknown, Problem>(error)
-        .with(P.instanceOf(ApiError<T | CommonErrorCodes>), (error) => {
-          const problem = makeProblem(httpMapper(error), error);
-          logger.warn(makeProblemLogString(problem, error));
-          return problem;
-        })
-        // this case is to allow a passthrough of PROBLEM errors in the BFF
-        .with(
-          {
-            response: {
+    const genericProblem = makeProblem(500, genericError("Unexpected error"));
+
+    return match<unknown, Problem>(error)
+      .with(P.instanceOf(ApiError<T | CommonErrorCodes>), (error) => {
+        const problem = makeProblem(httpMapper(error), error);
+        logger.warn(makeProblemLogString(problem, error));
+        return problem;
+      })
+      .with(
+        /* this case is to allow a passthrough of Problem errors, so that
+           services that call other interop services can forward Problem errors
+           as they are, without the need to explicitly handle them */
+        {
+          response: {
+            status: P.number,
+            data: {
+              type: "about:blank",
+              title: P.string,
               status: P.number,
-              data: {
-                type: "about:blank",
-                title: P.string,
-                status: P.number,
+              detail: P.string,
+              errors: P.array({
+                code: P.string,
                 detail: P.string,
-                errors: P.array({
-                  code: P.string,
-                  detail: P.string,
-                }),
-                correlationId: P.string.optional(),
-              },
+              }),
+              correlationId: P.string.optional(),
             },
           },
-          (e) => {
-            const receivedProblem: Problem = e.response.data;
-            if (operationalMsg) {
-              logger.warn(operationalMsg);
-            }
+        },
+        (e) => {
+          const receivedProblem: Problem = e.response.data;
+          if (operationalMsg) {
+            logger.warn(operationalMsg);
+          }
+
+          if (problemErrorsPassthrough) {
             logger.warn(makeProblemLogString(receivedProblem, error));
             return e.response.data;
+          } else {
+            logger.warn(
+              makeProblemLogString(
+                genericProblem,
+                `${receivedProblem.title}: code: ${
+                  receivedProblem.errors.at(0)?.code
+                } - detail: ${receivedProblem.errors.at(0)?.detail}`
+              )
+            );
+            return genericProblem;
```

Comment:
If we receive an error in this format it means it's coming from an API client call and it was not caught - in this case, since `problemErorrsPassthrough` is false, we return a generic problem error but log, as original error, the data of the problem we received from the API call